### PR TITLE
terror: add toml encoding

### PR DESCRIPTION
**Pull request by overvenus**
Signed-off-by: Neil Shen <overvenus@gmail.com>

### CODE REVIEW

File: terror_encoding.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+// RenderJSON implements MarshaJSON and UnmarshaJSON for Error.
+type RenderJSON Error
+
+var _ json.Marshaler = RenderJSON{}
+var _ json.Unmarshaler = &RenderJSON{}
+
+// MarshalJSON implements json.Marshaler interface.
+func (rd RenderJSON) MarshalJSON() ([]byte, error) {
```
Comment (imtbkcat) [inline terror_encoding.go:86]: here should be (rd *RenderJson)

File: terror_encoding.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	}
+	buf := new(bytes.Buffer)
+	if err := toml.NewEncoder(buf).Encode(render); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler interface.
+func (rd *RenderTOML) UnmarshalText(data []byte) error {
```
Comment (imtbkcat) [inline terror_encoding.go:60]: Why not `*Error` directly, but use `RenderTOML` and `RenderJSON`

File: terror_encoding.go (hunk 1 above)
Comment (overvenus) [reply terror_encoding.go]: It has been changed to function.

File: terror_encoding.go (hunk 2 above)
Comment (overvenus) [reply terror_encoding.go]: The UnmarshalText has been removed.

File: terror_encoding.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	return errorRender{
+		Msg:         e.GetMsg(),
+		Description: e.description,
+		Workaround:  e.workaround,
+		RFCCode:     string(e.codeText),
+	}
+}
+
+// RenderJSON implements MarshaJSON and UnmarshaJSON for Error.
+func RenderJSON(e Error) interface{} {
```
Comment (lonng) [inline terror_encoding.go:44]: Has anywhere called this function?

**Review by lonng — COMMENTED** [review]
Rest LGTM

**Review by imtbkcat — APPROVED** [review]
LGTM

**Pull request closed without merging**